src/base_vec.rs

```diff
@@ -183,7 +184,7 @@ impl<T: Storable, M: Memory> BaseVec<T, M> {
         let data_offset = self
             .write_entry_size(offset, bytes.len() as u32)
             .expect("unreachable: cannot fail to write to pre-allocated area");
-        self.memory.write(data_offset, bytes.borrow());
+        write(&self.memory, data_offset, bytes.borrow());
```

Comment:
Note: if we want to be consistent with other APIs in this data structure, we would've used `safe_write` here and then `set` would return a `Result<(), GrowFailed>`. This would be a breaking change that would affect both `StableVec` and `StableMinHeap`, and I didn't feel like it was worth the effort/hassle given that the benefit of returning a `GrowFailed` error is still questionable. In fact, I'm wondering if we should remove the `GrowFailed` error from all the other APIs given that it hasn't been that useful to my knowledge, and just makes code more verbose.

Reply:
I'd be in favour of removing as I'm also not sure what can a canister reasonably do if it hits the `GrowFailed` case. What course of action would you take? If you were able to run some logic when getting close to out of memory (aka having some hook) that could be useful but if you hit this error, you might as well trap yourself.

Reply:
Exactly. I think if the course of action to take would be, for example, spawn another canister to move your data, you'd want to do it well before you hit the grow limit. I'll create a ticket, and we can take care of this separately.